compiler/AST/symbol.cpp

```diff
@@ -606,7 +549,9 @@ GenRet VarSymbol::codegen() {
   ret.chplType = typeInfo();
 
   if( outfile ) {
-    if (immediate) {
+    // dtString immediates don't actually codegen as immediates, we just use
+    // them for param string functionality.
```

Comment:
Hmm... I think this is not the choice I would have made.
Couldn't you make a string immediate codegen as a call to some kind of string-build out of a c string?
Even if you want to write that code in Chapel, I think you could mark a relevant routine as 'export' and
then call it from this point in codegen. Or (better) you could transform string literals into calls of
"string build" at some earlier point - possibly in AST build. That would just return an appropriate
string record with owned=false and the pointer to the C string. Wouldn't filling in a record from a (local)
C string be faster than fishing out an initialized record from globals, possibly from Locale 0?
I'd like to discuss this further with you...